src/v/cloud_storage/remote_partition.cc

```diff
@@ -624,6 +624,31 @@ void remote_partition::return_reader(
     }
 }
 
+/// Return reader back to segment_state
+void remote_partition::evict_reader(
+  std::unique_ptr<remote_segment_batch_reader> reader) {
+    if (reader->is_eof()) {
+        auto offset = reader->base_rp_offset();
+        auto it = _segments.find(offset);
+        if (
+          it != _segments.end()
+          && reader->reads_from_segment(*it->second->segment)
+          && it->second->segment.use_count() == 2) {
+            // The segment is only used by the evicted reader.
+            // Normally, this can happen during batch processing, when
+            // the client scans the partition. In this case there is no
+            // contention and there is no point in keeping materialized
+            // segment around. The segment can still be cached in SI cache.
```

Comment:
Just making sure I understand this:  we’re not actually evicting from the cloud cache -- the files still remain on disk until the cache_service cleans it up, and the PR only cleans up the segment readers, right?